app/controllers/stats_controller.rb

```diff
@@ -0,0 +1,573 @@
+        ,coalesce (d.item_id, g.item_id) as transaction_id
+        ,coalesce(d.amount/100, g.giftcard_amount/100) as dollar_amount
+        ,date(i.created_at at time zone 'utc' at time zone 'America/New_York') as created
+        ,date(i.updated_at at time zone 'utc' at time zone 'America/New_York') as last_update
+        ,i.created_at at time zone 'utc' at time zone 'America/New_York' as created_timepstamp
+        ,i.refunded
+        from items as i 
+        left join donation_details as d
+        on i.id = d.item_id
+        left join giftcards as g      
+        on i.id = g.item_id
+        left join sellers as s 
+        on i.seller_id = s.id
+        left join contacts as c   
+        on i.purchaser_id = c.id
+        left join payment_intents as p
+        on i.payment_intent_id = p.id 
+        where refunded = 'false'
+        order by created_timepstamp desc
+        ), 
+        pool as (
+        select 
+            payment_intent_id
+            ,refunded
+            ,date(p.created_at at time zone 'utc' at time zone 'America/New_York') as created
+            ,count(*) as num
+        from donation_details as d
+        join items as i
+        on d.item_id = i.id
+        join payment_intents as p
+        on i.payment_intent_id = p.id 
+        group by 1,2,3
+        )
+    
+    -- -- RUN TO GET BI-WEEKLY PAYOUT $
+        SELECT  
+        -- merchant
+        min(created) as start_date
+        ,max(created) as end_date 
+        ,sum(case when purchase_type = 'donation' then dollar_amount else 0 end) as donation
+        ,sum(case when purchase_type = 'giftcard' AND gam is FALSE then dollar_amount else 0 end) as giftcard
+        ,sum(case when purchase_type = 'giftcard' AND gam is TRUE then dollar_amount else 0 end) as GAM
+        /*if purchase type is giftcard, and gam is TRUE then giftcard dollar amount? = GAM */ 
+        ,sum(dollar_amount) as total
+    
+        from all_donations
+        /*** Input Dates Here ***/ 
+        -- where created >= '2020-03-16' and created < '2021-03-23'
+        -- group by 
+        -- merchant
+        /*order by 1 */
+  values
+  
+          SQL
+        end 
+        def show2(donation_totals, sellers_total, transaction_totals, gam_count)
+            # puts (donation_totals)
+            # pry
+            
+            box1 =  "$%s" % ActionController::Base.helpers.number_with_precision( donation_totals.getvalue(0,5), :precision => 0, :delimiter => ',')  #"$10,000"
+            box2 =  ActionController::Base.helpers.number_with_precision( gam_count, :precision => 0, :delimiter => ',') 
+            box3 = "36,573" 
+            box4 = ActionController::Base.helpers.number_with_precision( transaction_totals, :precision => 0, :delimiter => ',') 
+            box5 = sellers_total
+            box6 = "$47,689"
+            render html:  <<-HERDOC
+            <div class="" data-sr-id="3"
```

Comment:
Question, and maybe we can have larger offline discussion -- do you think we should be having the backend return full html, or have the front-end have it and fill in the few data fields? We have this draft PR https://github.com/sendchinatownlove/sendchinatownlove.github.io/pull/499 from Matt, not sure if you started from there.

If we do prefer to have the backend return html  -- can we have these somewhere as html files, interpolate, then return?

Also, any reason this is `show2`, what happened to `show`? ahha

Reply:
> Question, and maybe we can have larger offline discussion -- do you think we should be having the backend return full html, or have the front-end have it and fill in the few data fields? We have this draft PR sendchinatownlove/sendchinatownlove.github.io#499 from Matt, not sure if you started from there.

So i thought this was also the best method, however how the app serves, getting it to render in an iframe without app headers and other attributes was very difficult. The HTML here is perfect because its a direct replication from the weebly site. 

If there was a way to hide both the app footer/header it would be easy to build a standalone component but it wasn't apparent. I pulled the latest version and ran it and it suffered from the framing issue. IE iframes only let you 100% the full view not part of it. 


> If we do prefer to have the backend return html -- can we have these somewhere as html files, interpolate, then return? 

this is fine! i ended up using HEREDOCS because i could not figure out the rails rendering engine documentation. 

I used for example 

`render stats/index` or the name of the file with an associated `view` and didn't get it working. I wasn't sure if there was a missing middleware I was missing somewhere since we aren't loading views any else in the application. 

I have no idea what happened to show1.. they must have left and never came back, ill update this to show.

Reply:
Thanks for the in-depth answers to all of these -- sounds like there are some deeper challenges we have to overcome -- thanks for taking it as far as it is and I think we should definitely get this merged now. If we learn how to do things better or what missing middlewares could help we can always refactor!